packages/hardhat-plugin/test/use-module.ts

```diff
@@ -1,157 +1,82 @@
 /* eslint-disable import/no-unused-modules */
-import { buildModule, IDeploymentBuilder } from "@ignored/ignition-core";
+import { defineModule } from "@ignored/ignition-core";
 import { assert } from "chai";
 
-import { mineBlocks } from "./helpers";
-import { useEnvironment } from "./useEnvironment";
+import { useEphemeralIgnitionProject } from "./use-ignition-project";
 
-describe.skip("useModule", function () {
-  useEnvironment("minimal");
+describe("useModule", function () {
+  useEphemeralIgnitionProject("minimal-new-api");
 
   describe("returning futures from module usage", () => {
     it("using useModule", async function () {
       await this.hre.run("compile", { quiet: true });
 
-      const thirdPartyModule = buildModule(
-        "ThirdPartySubmodule",
-        (m: IDeploymentBuilder) => {
-          const foo = m.contract("Foo");
+      const thirdPartyModule = defineModule("ThirdPartySubmodule", (m) => {
+        const foo = m.contract("Foo");
 
-          return { foo };
-        }
-      );
+        return { foo };
+      });
 
-      const userModule = buildModule("UserModule", (m: IDeploymentBuilder) => {
+      const userModule = defineModule("UserModule", (m) => {
         const { foo } = m.useModule(thirdPartyModule);
 
-        m.call(foo, "inc", {
-          args: [],
-        });
+        m.call(foo, "inc");
 
         return { foo };
       });
 
-      const deployPromise = this.hre.ignition.deploy(userModule, {
-        parameters: {},
-        ui: false,
-      });
-
-      await mineBlocks(this.hre, [1, 1, 1], deployPromise);
-
-      const result = await deployPromise;
-
-      assert.isDefined(result);
-
-      const x = await result.foo.x();
+      const result = await this.deploy(userModule);
 
-      assert.equal(x, Number(2));
+      assert.equal(await result.foo.x(), Number(2));
     });
   });
 
-  describe("passing futures into submodules", () => {
-    it("using useModule", async function () {
+  describe("modules depending on modules", () => {
+    // TODO: confirm we are happy for this to go
```

Comment:
Ordering of `useModule` is now not expressable. Just wanting to double check we have remembered that before I delete this test. @alcuadrado

Reply:
Yes. Expected behavior.

Reply:
If we want to add it later, it can be added as a batching rule. Similar to the malaga rule.